packages/good-design/src/apps/goodid/screens/OnboardScreen.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Container, Heading, HStack, Text, VStack } from "native-base";
+import { useIdentityExpiryDate, useIsAddressVerified } from "@gooddollar/web3sdk-v2";
+import { noop } from "lodash";
+import moment from "moment";
+
+import { useFVModalAction } from "../../../hooks/useFVModalAction";
+import { withTheme } from "../../../theme";
+import { Title, TxModal } from "../../../core";
+import { BaseButton } from "../../../core/buttons";
+import { GoodIdCard } from "../idcard";
+import SvgXml from "../../../core/images/SvgXml";
+import DollarSvg from "../../../assets/svg/goodid/dollar.svg";
+import GlobusSvg from "../../../assets/svg/goodid/globus.svg";
+import HeartsSvg from "../../../assets/svg/goodid/hearts.svg";
+import StopWatchSvg from "../../../assets/svg/goodid/stopwatch.svg";
+import UbiSvg from "../../../assets/svg/goodid/ubi.svg";
+
+interface OnboardScreenProps {
+  account: string | undefined;
+  navigateTo?: () => void;
+  innerContainer?: any;
+  fontStyles?: any;
+}
+
+const accessList = [
+  {
+    label: "Crypto UBI",
+    icon: UbiSvg
+  },
+  {
+    label: "Humanitarian funds",
+    icon: HeartsSvg
+  },
+  {
+    label: "Climate relief disbursements",
+    icon: GlobusSvg
+  },
+  {
+    label: "Financial services",
+    icon: DollarSvg
+  }
+];
+
+const OnboardScreen = withTheme({ name: "OnboardScreen" })(
+  ({ navigateTo, account, innerContainer, fontStyles, ...props }: OnboardScreenProps) => {
+    const [isWhitelisted] = useIsAddressVerified(account ?? "");
+    const [expiryDate] = useIdentityExpiryDate(account ?? "");
+    const [formattedExpiryDate, setExpiryDate] = React.useState<string | undefined>();
+    const [isPending, setPendingSignTx] = useState(false);
+    const { title, listLabel, poweredBy, tos } = fontStyles ?? {};
+
+    const { verify } = useFVModalAction({
+      firstName: "Test", //todo: is this value required or can it be made optional?
+      method: "redirect",
+      chainId: 42220,
+      onClose: noop
+    });
+
+    const handleShouldFv = useCallback(async () => {
+      const { expiryTimestamp } = expiryDate || {};
+
+      if (isWhitelisted && expiryTimestamp) {
+        const expiry = moment(expiryTimestamp.toNumber());
+        const threeMonthsFromNow = moment().clone().add(3, "months");
+        const isWithinThreeMonths = expiry.isBefore(threeMonthsFromNow);
+
+        if (isWithinThreeMonths) {
+          navigateTo ? navigateTo() : await verify();
+        } else {
+          //todo: Need solution for widget-navigation (gooddapp)
```

Comment:
as described in PR description, this will be added/handled when this issue is done: https://github.com/GoodDollar/GoodWeb3-Mono/issues/131

Reply:
you should avoid putting logic in good-design.
ie accept to the component a callback that decides where to send the user to

Reply:
The only reason the alternative flow exists is because of the wallet in-app navigation

any other implementation and a alternative navigation flow doesn't make sense (to me)
why I made it part of the screen itself

Reply:
the component should accept a callback